### ISSUE TITLE
Slack API `files.upload` v1 deprecation

### ISSUE THREAD

**Issue by asaf400** [apache/superset#28353]
### Bug description

The API superset uses for slack reports has been deprecated, and needs to be replaced with the new API:

[As described here](https://api.slack.com/methods/files.upload)
`files.upload is deprecated and will stop functioning on March 11, 2025. Use [files.getUploadURLExternal](https://api.slack.com/methods/files.getUploadURLExternal) and [files.completeUploadExternal](https://api.slack.com/methods/files.completeUploadExternal) to upload files instead. Newly created apps will be unable to use files.upload beginning May 8, 2024. See [Uploading files](https://api.slack.com/messaging/files#uploading_files) for more details on the process and [this changelog](https://api.slack.com/changelog/2024-04-a-better-way-to-upload-files-is-here-to-stay) for more on the deprecation.
`

Seems like new slack-sdk version required minimum is [v3.19.0](https://github.com/slackapi/python-slack-sdk/releases/tag/v3.19.0)
Which Superset already uses https://github.com/apache/superset/blob/f5843fe588112c5c303215945a36fda0845c9e99/pyproject.toml#L88

```
[root@LAPTOP-ASAF-T14 ~]# pip install 'slack_sdk>=3.19.0, <4'
Collecting slack_sdk<4,>=3.19.0
...
Installing collected packages: slack_sdk
Successfully installed slack_sdk-3.27.1
```

additional resources: 
https://github.com/slackapi/python-slack-sdk/pull/1272

### How to reproduce the bug

Deprecation

### Screenshots/recordings

As a slack organization administrator I received the following message by email:
 
<img width="422" alt="2024-05-06 12_40_45-Gmail - Upcoming changes to support for the files upload API method - asaf levy@" src="https://github.com/apache/superset/assets/14169006/d61cd2b6-50f3-4896-b361-7f3ba98664c3">

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [x] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [x] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [x] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by mistercrunch** [comment]
https://github.com/apache/superset/pull/28423 should do it, not sure how to test/validate though

**Comment by eschutho** [comment]
I think the above fix should cover this issue. Thanks for the report @asaf400!

**Comment by MeD1100** [comment]
I've deployed superset using the helm chart. I'm currently using app v3.1.0 and chart v0.12.3
I'm facing the same issue. Do I need to update/change something from my side so that the slack deprecation is fixed? Please help me out.
Thanks in advance!

**Comment by isharamet** [comment]
Just FYI, after switching to `file_upload_v2` you won't be able to specify channel name, only it's ID due to the following issue:
https://github.com/slackapi/python-slack-sdk/issues/1326

**Comment by PeterGrace** [comment]
> The fix is already in master - merged after v4.0.1, so just wait for the next release build, or locally build `master`

@asaf400 I just deployed 4.0.2 a few minutes ago and am getting this error still.  Did this fix make it into 4.0.2 appropriately?  Is there a change to deployment documentation to deploy this fix?  A new feature flag, perhaps?

Thanks in advance for your assistance.

**Comment by rusackas** [comment]
CC @michael-s-molina regarding whether or not this was cherried into 4.0.2 (or could be cherried into 4.0.3).

**Comment by michael-s-molina** [comment]
@rusackas This was not included in 4.0.2 but will be in 4.1.0 which will be the next release of Superset @sadpandajoe @eschutho

**Comment by sadpandajoe** [comment]
> @rusackas This was not included in 4.0.2 but will be in 4.1.0 which will be the next release of Superset @sadpandajoe @eschutho

The first two attempts to fix the issue wasn't working well. @eschutho created this SIP: https://github.com/apache/superset/issues/29263 and is currently working on it.

**Comment by PeterGrace** [comment]
> @rusackas This was not included in 4.0.2 but will be in 4.1.0 which will be the next release of Superset @sadpandajoe @eschutho

> The first two attempts to fix the issue wasn't working well. @eschutho created this SIP: https://github.com/apache/superset/issues/29263 and is currently working on it.

What is the timeframe for this?  As far as I can see, slack notifications are completely inoperable at this point unless one builds from source removing the file upload from the message?

**Comment by sadpandajoe** [comment]
> > @rusackas This was not included in 4.0.2 but will be in 4.1.0 which will be the next release of Superset @sadpandajoe @eschutho
> 
> > The first two attempts to fix the issue wasn't working well. @eschutho created this SIP: #29263 and is currently working on it.
> 
> What is the timeframe for this? As far as I can see, slack notifications are completely inoperable at this point unless one builds from source removing the file upload from the message?

You can follow this PR: https://github.com/apache/superset/pull/29264 to see when it gets merged.
